src/clj_gatling/simulation_runners.clj

```diff
@@ -11,21 +11,25 @@
 (deftype DurationRunner [^Duration duration]
   RunnerProtocol
   (calculate-progress [_ _ start now]
-    (let [time-taken (Duration/between ^LocalDateTime start ^LocalDateTime now)
-          time-taken-in-millis (.toMillis time-taken)
-          duration-in-millis (max (.toMillis duration) 1)]
-      ;; Progress should not exceed 1.0 eventhough more than max duration might pass before simulation finishes
-      [(min 1.0 (float (/ time-taken-in-millis duration-in-millis))) time-taken]))
+    (if (or (nil? start) (nil? now))
+      [0.0 (Duration/ZERO)]
+      (let [time-taken (Duration/between ^LocalDateTime start ^LocalDateTime now)
+            time-taken-in-millis (.toMillis time-taken)
+            duration-in-millis (max (.toMillis duration) 1)]
+        ;; Progress should not exceed 1.0 even though more than max duration might pass before simulation finishes
+        [(min 1.0 (float (/ time-taken-in-millis duration-in-millis))) time-taken])))
   (continue-run? [_ _ start next]
-    (.isBefore ^LocalDateTime next (.plus ^LocalDateTime start duration)))
+    (or (nil? start) (.isBefore ^LocalDateTime next (.plus ^LocalDateTime start duration))))
   (runner-info [_] (str "duration " duration)))
 
 (deftype FixedRequestNumberRunner [number-of-requests]
   RunnerProtocol
   (calculate-progress [_ sent-requests start now]
-    (let [time-taken (Duration/between ^LocalDateTime start ^LocalDateTime now)]
-      ;; Progress should not exceed 1.0 eventhough more than max requests might be sent before simulation finishes
-      [(min 1.0 (float (/ sent-requests number-of-requests))) time-taken]))
+    (if (or (nil? start) (nil? now))
+      [0.0 (Duration/ZERO)]
+      (let [time-taken (Duration/between ^LocalDateTime start ^LocalDateTime now)]
+        ;; Progress should not exceed 1.0 even though more than max requests might be sent before simulation finishes
+        [(min 1.0 (float (/ sent-requests number-of-requests))) time-taken])))
```

Comment:
This is all to handle the fact that the value of `start` will be `nil` this is the first request to be sent.

Reply:
Yeah. Makes sense